File: leetcode/1509/MinDifference.cs
public class Solution
{
    public int MinDifference(int[] nums)
    {
        if (nums.Length <= 4) return 0;
        int min = Int32.MaxValue;

        Array.Sort(nums);
        for (int i = 0; i <= 3; i++)
        {
            min = Math.Min(min, nums[nums.Length - 1 - i] - nums[3 - i]);
        }

        return min;
    }
}

File: leetcode/1509/MinDifference.java
import java.util.*;

class minDifference {
    public static void main(String[] args) {
        Solution sl = new Solution();
        System.out.println(sl.minDifference(new int[] { 1, 5, 0, 10, 14 }));
    }
}

class Solution {
    public int minDifference(int[] nums) {
        if (nums.length <= 4)
            return 0;
        int min = Integer.MAX_VALUE;

        Arrays.sort(nums);
        for (int i = 0; i <= 3; i++) {
            min = Math.min(min, nums[nums.length - 1 - i] - nums[3 - i]);
        }

        return min;
    }
}